src/content-script/listeners/index.ts

```diff
@@ -11,10 +12,18 @@ import '@contentScript/listeners/player';
  * @see https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage
  */
 window.addEventListener('message', event => {
+  const storeState = store.getState();
   switch (event.data.name) {
     case WindowMessages.URL_CHANGE:
       debug(event.data.name);
       debug(document.location.href);
+      // check to see if user is host before switching URLS
+      if (storeState.party.isHost && storeState.party.roomId != null) {
+        store.dispatch(setRoomId(storeState.party.roomId));
+        const newURL = `${document.location.href}&couchSyncRoomId=${storeState.party.roomId}`;
+        socket.emit(SocketEvents.URL_CHANGE, `URL Changed in room ${storeState.party.roomId}`);
+        store.dispatch(setJoinUrl(newURL)); // this is kinda hacky but it works
+      }
```

Comment:
- Can we export into a function. There are more cases we might need to handle (socket teardown if not host) and we don't want all the logic to live in this block.
- Can you use `query-string` to generate the new join_url? See https://github.com/rkrishn7/playback_sync_api/blob/main/src/lib/services/party.ts#L16